packages/@aws-cdk/aws-stepfunctions-tasks/lib/run-batch-job.ts

```diff
@@ -0,0 +1,341 @@
+import * as batch from '@aws-cdk/aws-batch';
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import { Aws, Duration } from '@aws-cdk/core';
+import { getResourceArn } from './resource-arn-suffix';
+
+/**
+ * An object representing an AWS Batch array job.
+ */
+export interface ArrayProperties {
+  /**
+   * The size of the array job.
+   *
+   * @default - No size
+   */
+  readonly size?: number;
+}
+
+/**
+ * A key-value pair object.
+ */
+export interface KeyValuePair {
+  /**
+   * The name of the key-value pair.
+   * For environment variables, this is the name of the environment variable.
+   *
+   * @default - No name
+   */
+  readonly name?: string;
+
+  /**
+   * The value of the key-value pair.
+   * For environment variables, this is the value of the environment variable.
+   *
+   * @default - No value
+   */
+  readonly value?: string;
+}
+
+/**
+ * The type and amount of a resource to assign to a container.
+ * Currently, the only supported resource type is GPU.
+ */
+export interface ResourceRequirement {
+  /**
+   * The type of resource to assign to a container.
+   * Currently, the only supported resource type is GPU.
+   */
+  readonly type: string;
+
+  /**
+   * The number of physical GPUs to reserve for the container.
+   * The number of GPUs reserved for all containers in a job
+   * should not exceed the number of available GPUs on the compute
+   * resource that the job is launched on.
+   */
+  readonly value: string;
+}
+
+/**
+ * The overrides that should be sent to a container.
+ */
+export interface ContainerOverrides {
+  /**
+   * The command to send to the container that overrides
+   * the default command from the Docker image or the job definition.
+   *
+   * @default - No command overrides
+   */
+  readonly command?: string[];
+
+  /**
+   * The environment variables to send to the container.
+   * You can add new environment variables, which are added to the container
+   * at launch, or you can override the existing environment variables from
+   * the Docker image or the job definition.
+   *
+   * @default - No environment overrides
+   */
+  readonly environment?: KeyValuePair[];
+
+  /**
+   * The instance type to use for a multi-node parallel job.
+   * This parameter is not valid for single-node container jobs.
+   *
+   * @default - No instance type overrides
+   */
+  readonly instanceType?: string;
```

Comment:
I think it might be `string`. I verified it in the [SDK](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/batch/model/ContainerOverrides.html) as well.

Reply:
See the implementation in Batch's own APIs - https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-batch/lib/job-definition.ts#L40